### DISCUSSION TITLE
Update a DateTime field value with the current server datetime

### DISCUSSION THREAD

**Discussion by oscar-costa** [pocketbase/pocketbase#4411] [Q&A] [answered]
When updating a record, is it possible to update a particular DateTime field value with the current server datetime, and not be dependent on a value send by the client?

**Comment by starredev** [comment]
It’s always in UTC and not changeable. It makes no sense because the browser (frontend) always takes the current timezone for the user.

**Comment by ganigeorgiev** [comment, marked as answer]
Only if you use the hooks ([Go](https://pocketbase.io/docs/go-event-hooks/) or [JS](https://pocketbase.io/docs/js-event-hooks/)).

- Go:
    ```go
    app.OnRecordBeforeUpdateRequest("yourCollection").Add(func(e *core.RecordUpdateEvent) error {
        e.Record.Set("yourDateField", types.NowDateTime())

        return nil
    })
    ```

- JS `pb_hooks`:
    ```js
    onRecordBeforeUpdateRequest((e) => {
        e.record.set("yourDateField", new Date())
    }, "yourCollection")
    ```

**Reply by oscar-costa** [reply]
In the onRecordBeforeUpdateRequest() event, is it possible to get a specific existing field value of the record, before the update action, so we can compare the old value with the new one?

**Reply by ganigeorgiev** [reply]
@oscar-costa Yes, please refer to:

- Go: https://pocketbase.io/docs/go-records/#getset-record-fields
- JS: https://pocketbase.io/docs/js-records/#getset-record-fields

For example, if you are using the JS `pb_hooks` to get the old/original value you can call `e.record.originalCopy().get("yourDateField")`.
